### ISSUE TITLE
speed of LA4Falcon?

### ISSUE THREAD

**Issue by dgordon562** [PacificBiosciences/FALCON#150]
My v.0.3.0 LA4Falcon jobs are running about 6 or 8 times longer (gasp!) than v.0.2.2 LA4Falcon.  Is this to be expected?  (24 hours to complete vs a mean of 3.5 hours to complete.)

If not, I will get more data--trying both versions on the same las files in a more controlled test.  I'm also wondering if I have network problems now.

But I thought I would first ask if you know the relative performance?

**Comment by pb-jchin** [comment]
Is it the same coverage data and parameter? The consensus algorithm could be slower than before but it won't always call missing base for het-SNP site anymore.  In my test of 50 to 60x human haploid data, the consense step takes about 1.3 k cpu hours and about 4.0x faster with more alginment parameters which ignores most of small alignments between repeats.

**Comment by dgordon562** [comment]
size of las files 
v0.3.0 assembly:  median of about 7.1 MB
v0.2.2 assembly: median of about 4.1 MB

v0.2.2:
length_cutoff = 8000
length_cutoff_pr = 6000
falcon_sense_option = --output_multi --min_idt 0.70 --min_cov 4 --local_match_count_threshold 2 --max_n_read 200 --n_core 6
overlap_filtering_setting = --max_diff 60 --max_cov 60 --min_cov 2 


v0.3.0
length_cutoff = 10000
length_cutoff_pr = 6000
falcon_sense_option = --output_multi --min_idt 0.70 --min_cov 4 --local_match_count_threshold 2 --max_n_read 200 --n_core 6 --output_dformat
overlap_filtering_setting = --max_diff 100 --max_cov 100 --min_cov 2 --bestn 10 --n_core 24


identical:
pa_HPCdaligner_option =  -v -dal128 -t16 -e.70 -l1000 -s1000
ovlp_HPCdaligner_option = -v -dal128 -t32 -h60 -e.96 -l500 -s1000
pa_DBsplit_option = -x500 -s400
ovlp_DBsplit_option = -x500 -s400

I'm wondering if the problem might be the overlap_filtering_setting?
If so, is there a way I could restart it without redoing all of the
daligner steps?

**Comment by dgordon562** [comment]
Jason:  my question was about the overlap_filtering_setting:
I'm wondering if the problem might be the overlap_filtering_setting?
If so, is there a way I could restart it without redoing all of the daligner steps?

There is a loop in LA4Falcon that goes around about 900 million times.  I inserted code to show me the iterations of that loop.  Often it doesn't go around, using no cpu time and being in sleep state, but fc_consensus.py is using 100% cpu.  There is no longer any evidence of waiting on disk.  Does this behavior make sense to you?

**Comment by dgordon562** [comment]
Could I have an answer, please?

**Comment by pb-jchin** [comment]
`overlap_filtering_setting` is not used during the consensus stage. it is used after the 2nd overlaps. It appears to me your 2nd genome is more repetitive.  You will need to optimize the `pa_HPCdaligner_option` to reduce the some of the reduant overlap caused by repeats. I have discussed this issue with memebrs of your lab. Maybe they can help you.  If you don't want to change the parameters, then those repeats can be a problem. I have terminted some assembly like that and re-think the problem more careful to solve it.
